ps5Controller/ps5Controller/HomeModule/HomeInteractor.swift

```diff
@@ -0,0 +1,19 @@
+protocol HomeInteractorInterface {
+    func fetchProductList(type: HomeTabType)
+}
+
+final class HomeInteractor: HomeInteractorInterface {
+    var presenter: HomePresenter?
```

Comment:
komple presenter'ın tüm methodlarına erişen bi variable'ı burda tutmak nasıl bi yaklaşım? open closed prensibine uyuyor mu? 

şöyle bi şey yapabilirsin. presenter tutmak yerine output denen bi variable'ın olur. Yeni bi protocol'den türer. Ve sadece içinde `productListFetched` methodun olur. Bunu presenter conform eder. Sen nihayetinde yine presenter'ı paslamış olursun ama onun scope'unu `HomeInteractorOutput` adını verdiğimiz bi protokol belirler.
Hem burada retain cycle olmaması için weak var yaratman gerekiyor. 🚀